### PR TITLE
Add 'None' option to print configuration fighter selection

### LINKED ISSUE #1094: Print Configuration: Fighter Selection needs a "None" option
Right now: our JS logic insists on either "All current & future fighters" or one-or-more fighters being picked. This is a problem now that we have blank cards — you might want to an entirely blank print config.

We should add a "None" option under "All current & future fighters" which will simply print none of the fighters.

### PR DESCRIPTION
**Pull request by tgvashworth**
This change adds a new fighter selection mode to print configurations, allowing users to create print configs that include only blank cards without any fighters.

Changes:
- Added fighter_selection_mode field to PrintConfig model with three options:
  * 'all': All current & future fighters (default)
  * 'specific': Select specific fighters
  * 'none': No fighters (new option)
- Updated PrintConfigForm to use radio buttons instead of checkbox
- Updated form template with new radio button UI
- Added JavaScript to enable/disable fighter checkboxes based on mode
- Updated ListPrintView to handle NO_FIGHTERS mode
- Created migration for the new field

This enables users to print entirely blank card sets, which is useful when they want blank cards without any existing fighters.

Fixes #1094

🤖 Generated with [Claude Code](https://claude.ai/code)

### CODE REVIEW

File: gyrinx/core/views/list.py [hunk 1]
```diff
@@ -405,14 +405,25 @@ def get_context_data(self, **kwargs):
...
+                # No fighters - return empty queryset
+                fighters_qs = fighters_qs.none()
+            elif print_config.fighter_selection_mode == PrintConfig.SPECIFIC_FIGHTERS:
+                # Filter by included fighters if specific ones are selected
+                if print_config.included_fighters.exists():
+                    fighters_qs = fighters_qs.filter(
+                        id__in=print_config.included_fighters.values_list(
+                            "id", flat=True
+                        )
+                    )
```
[the remark this answers is not in the document]
Comment (tgvashworth) [reply gyrinx/core/views/list.py]: Fixed — added `else: fighters_qs = fighters_qs.none()` for the SPECIFIC_FIGHTERS case with no selections.

File: gyrinx/core/models/print_config.py [hunk 2]
```diff
@@ -95,13 +113,17 @@ def card_summary(self):
...
+        # Handle fighter selection based on mode
+        if self.fighter_selection_mode == self.NO_FIGHTERS:
+            included.append("No Fighters")
+        elif self.fighter_selection_mode == self.SPECIFIC_FIGHTERS:
+            fighter_count = self.included_fighters.count()
+            if fighter_count > 0:
+                included.append(
+                    ngettext("%(count)d Fighter", "%(count)d Fighters", fighter_count)
+                    % {"count": fighter_count}
+                )
```
[the remark this answers is not in the document]
Comment (tgvashworth) [reply gyrinx/core/models/print_config.py]: Fixed — now shows "0 Fighters" in the summary for this edge case.

File: gyrinx/core/tests/test_print_config_stash.py [hunk 3]
```diff
@@ -143,3 +144,48 @@ def test_print_view_default_shows_stash(authenticated_client, test_list_with_sta
...
+            "name": "Test Config",
+            "fighter_selection_mode": PrintConfig.SPECIFIC_FIGHTERS,
+            "included_fighters": [regular_fighter.id],
+            "blank_fighter_cards": 0,
+            "blank_vehicle_cards": 0,
+        },
+        list_obj=test_list,
+    )
+
+    assert form.is_valid()
```
[the remark this answers is not in the document]
Comment (tgvashworth) [reply gyrinx/core/tests/test_print_config_stash.py]: Noted — not adding this for now.

File: gyrinx/core/tests/test_print_config_stash.py [hunk 4]
```diff
@@ -143,3 +144,48 @@ def test_print_view_default_shows_stash(authenticated_client, test_list_with_sta
...
     # By default (no print_config), stash should be shown
     assert "Stash" in response.content.decode()
+
+
+@pytest.mark.django_db
+def test_specific_fighters_requires_selection(
+    authenticated_client, test_list_with_stash
+):
+    """Test that 'Specific fighters' mode requires at least one fighter selected."""
+    client, user = authenticated_client
```
[the remark this answers is not in the document]
Comment (tgvashworth) [reply gyrinx/core/tests/test_print_config_stash.py]: Fixed.
